programs/survival/rope_ladders.sc

```diff
@@ -4,27 +4,88 @@
 // (Carpet Mod 1.4.9)
 //
 // Right clicking on a ladder with an other one, will extend the existing one down.
-///
+
+// 'Updated' by opsaaaaa for 1.19
+// added break unsuported ladder fearue
 
 __config() -> {'stay_loaded'->true};
 
-_rope_ladders(player, item_tuple, hand, block, face) -> (
-    g = player ~ 'gamemode';
-    if(g == 'spectator' || !item_tuple, return());
-    [item, count, nbt] = item_tuple;
-    if(item!='ladder' || block!='ladder' || property(block, 'facing') != face,return());
-    if(g=='adventure' && !nbt:'CanPlaceOn' ~ '"minecraft:ladder"', return());
-    b = neighbours(block):1;
-    if(b == 'ladder' && property(b,'facing') == property(block, 'facing'),
-        _rope_ladders(player, item_tuple, hand, b, face);
-        return()
+
+
+__on_player_right_clicks_block(p, item, hand, block, face, hitvec) -> (
+  if(_rope_should_start(p, item, block),
+    if( _attempt_place_ladder(_loop_to_ladder_base(block), block) &&
+        p~'gamemode' != 'creative',
+
+      _useup_ladder(p, item, hand);
+    );
+  );
+);
+
+__on_player_breaks_block(p, ladder) -> (
+  if(ladder=='ladder',
+    _break_unsuported_ladders(_under_ladder(ladder));
+  );
+);
+
+
+_break_unsuported_ladders(ladder) -> (
+  if(ladder == 'ladder',
+    if(!_support_is_solid(ladder),
+      destroy(ladder);
+      schedule(1,'_break_unsuported_ladders', _under_ladder(ladder));
     );
-    if(!air(b) && b != 'water', return());
-    set(b, block, 'waterlogged', if(b == 'water', 'true', 'false'));
-    if(g == 'creative', return()); 
-    inventory_set(player, if(hand=='mainhand',player~'selected_slot',-1), count - 1, item ,nbt)
+  );
+);
+
+
+_rope_should_start(p, item, block) -> (
+    p~'gamemode' != 'spectator' &&
+    item && item:0 == 'ladder' &&
+    block~'ladder' &&
+    if(g=='adventure', nbt:'CanPlaceOn'~'"minecraft:ladder"', true)
 );
 
-__on_player_right_clicks_block(player, item_tuple, hand, block, face, hitvec) -> (
-    _rope_ladders(player, item_tuple, hand, block, face)
-)
+
+_loop_to_ladder_base(block) -> (
+    b = _under_ladder(block);
+
+    if(b == 'ladder' && block_state(b,'facing') == block_state(block, 'facing'),
+
```

Comment:
This is to prevent the rope ladder from continuing a ladder below itself in the wrong orientation.
I could live without the block state check tbh.

Reply:
Do you think it would be possible to do this without recursion? Cos like imo a while loop fits here just fine, and so you can also add a limit, like the world height, in case something goes wrong.

Reply:
A while loop would likely work. I'll try it.